### PR TITLE
Add and delete load balancer in front of apiserver.

### PR DESCRIPTION
**Pull request by fgrzadkowski**
First PR to automate replicating the master in GCE.
#21124

@davidopp @roberthbailey @mikedanese @jszczepkowski @wojtek-t

### CODE REVIEW

**Comment by fgrzadkowski** [comment]
This is obviously only the first in automating HA master deployment, so maybe we should rename the flag to start with `EXPERIMENTAL`? WDYT?

File: cluster/gce/util.sh [hunk 1]
```diff
@@ -725,6 +729,74 @@ function create-master() {
...
+# Creates load balancer in front of apiserver if it doesn't exists already. Assumes there's only one
+# existing master replica.
+#
+# Assumes:
+#   PROJECT
+#   MASTER_NAME
+#   ZONE
+function create-loadbalancer() {
+  detect-master
+  local REGION=${ZONE%-*}
```
Comment (roberthbailey) [inline cluster/gce/util.sh]: Should we compute REGION once rather than in each function?

File: cluster/gce/util.sh [hunk 2]
```diff
@@ -725,6 +729,74 @@ function create-master() {
...
+  gcloud compute target-pools add-instances ${MASTER_NAME} --instances ${MASTER_NAME} --zone ${EXISTING_MASTER_ZONE}
+
+  # Step 3: Create forwarding rule.
+  # TODO: This step can take up to 20 min. We need to speed this up...
+  gcloud compute forwarding-rules create ${MASTER_NAME} \
+    --project "${PROJECT}" --region ${REGION} \
+    --target-pool ${MASTER_NAME} --address=${KUBE_MASTER_IP} --ports=443
+
+  echo -n "Waiting for the load balancer configuration to propagate..."
+  until $(curl -k -m1 https://${KUBE_MASTER_IP} > /dev/null 2>&1); do echo -n .; done
```
Comment (roberthbailey) [inline cluster/gce/util.sh]: This should have a timeout so it doesn't get stuck spinning forever if there is an error.

**Comment by roberthbailey** [comment]
Unless you plan to keep it called "Experimental" in 1.4 (when this is in the next stable release), I don't think calling it that now and then trying to change it later makes sense.

**Comment by roberthbailey** [comment]
Couple of nits, but nothing worth blocking this on. Please clean them up as you make forward progress in subsequent PRs.

**Comment by fgrzadkowski** [comment]
@piosz I have some follow up PRs that depend on this one. Can we merge this manually? As it is only in `kube-up` and `kube-down` it should not affect our e2e tests, if didn't break them completely.

**Comment by wojtek-t** [comment]
This PR broke all Jenkins suites - we are not removing IP addresses.

**Pull request merged**